src/assets/scripts/client/trafficGenerator/buildPreSpawnAircraft.js

```diff
@@ -94,8 +94,13 @@ const _calculateDistancesAlongRoute = (waypointModelList, airport) => {
 
     for (let i = 0; i < waypointModelList.length; i++) {
         const waypoint = waypointModelList[i];
-        const waypointPosition = waypoint.relativePosition;
         let previousWaypoint = waypoint;
+
+        if (waypoint.isVector || previousWaypoint.isVector) {
```

Comment:
I don't know if this PR is "ready enough" for this comment to be of any use but isn't the waypoint and previousWaypoint same here?